plugins/amqp_trx_plugin/amqp_trx_plugin.cpp

```diff
@@ -58,12 +88,39 @@ struct amqp_trx_plugin_impl : std::enable_shared_from_this<amqp_trx_plugin_impl>
          } else {
             FC_THROW_EXCEPTION( fc::out_of_range_exception, "Invalid which ${w} for consume of transaction_type message", ("w", which) );
          }
+         if( acked == ack_mode::received ) {
+            amqp_trx->ack( delivery_tag );
+         }
          return;
       } FC_LOG_AND_DROP()
 
       amqp_trx->reject( delivery_tag );
    }
 
+   void on_block_start( uint32_t bn ) {
+      trx_queue_ptr->on_block_start();
+   }
+
+   void on_block_abort( uint32_t bn ) {
+      if( acked == ack_mode::in_block ) {
+         tracked_delivery_tags.erase( bn );
+      }
+      trx_queue_ptr->on_block_stop();
+   }
+
+   void on_accepted_block( const chain::block_state_ptr& bsp ) {
+      if( acked == ack_mode::in_block ) {
+         const auto& entry = tracked_delivery_tags.find( bsp->block_num );
+         if( entry != tracked_delivery_tags.end() ) {
+            for( const auto& q : entry->second ) {
+               amqp_trx->ack( q );
```

Comment:
Doesn't AMQP have a batch ack?

Reply:
AMQP has an option to ack a delivery_tag and all delivery_tags before it. Since we are executing in order, can just store last delivery_tag per block and then ack just that last delivery_tag.